### DISCUSSION TITLE
Auto-map labels to project fields

### DISCUSSION THREAD

**Discussion by mrgrain** [community/community#60313] [Projects and Issues]
### Select Topic Area

Product Feedback

### Feature Area

Projects

### Body

PRs and Issues use labels to convey information and for filtering.
While Projects can display labels, they are not as flexibly to work with as fields, e.g. you cannot group by a label

I would like a field type that can map between labels and project fields.
I.e. if we an issue has the label `bug`, it the field `type` can will be assigned this value.

There will be a need for some logic to resolve conflicts, i.e. if an issue has both `bug` and `feature`.
I could see this mapping being like this:

```
Field: Type

Assign value "bug" when issue matches filter "label:bug -label:feature"
Assign value "feature" when issue matches filter "label:feature"
```

This would even support any kind of filter, not just labels.

Probably needs a setting to either stop evaluation after first match or after all rules.

### Guidelines

- [X] I have read the above statement and can confirm my post is relevant to the GitHub feature areas [Issues](https://docs.github.com/en/issues/tracking-your-work-with-issues/about-issues) and/or [Projects](https://docs.github.com/en/issues/planning-and-tracking-with-projects/learning-about-projects/about-projects).

**Comment by baynezy** [comment]
The lack of this functionality is why I have shied away from using projects thus far. I currently use labels to represent the status of a task as it is much easier to work with from the gh CLI. However, I would like to be able to sit at the browser level and do management of the project in the project view and have that keep both the labels and the project in sync.

**Comment by cschreib-ibex** [comment]
We recently migrated to using GitHub projects, and team members are used to setting labels, but not to set project fields. Project fields are "hidden away" on the issue view (click "+x more" to see them). Having labels mapped to project fields would solve these pain points. In our case, the label is a "priority" label, which has multiple possible values. There is a corresponding project field, so we can group/sort issues by priority in the project view. But right now, we need to duplicate the information in the label and the project field.
